functions/src/models/session.spec.ts

```diff
@@ -0,0 +1,332 @@
+import {mockFirebase} from 'firestore-jest-mock';
+
+mockFirebase(
+  {
+    database: {
+      sessions: [],
+    },
+  },
+  {includeIdsInData: true, mutable: true},
+);
+
+import {
+  mockCollection,
+  mockGetTransaction,
+  mockOrderBy,
+  mockRunTransaction,
+  mockUpdateTransaction,
+  mockWhere,
+  mockDelete,
+  mockDoc,
+} from 'firestore-jest-mock/mocks/firestore';
+import {firestore} from 'firebase-admin';
+import {Timestamp} from 'firebase-admin/firestore';
+
+import {
+  addSession,
+  deleteSession,
+  getSessions,
+  updateExerciseState,
+  updateSession,
+} from './session';
+import {SessionType} from '../../../shared/src/types/Session';
+
+const sessions = [
+  {
+    id: 'some-session-id',
+    name: 'some-name',
+    url: 'some-url',
+    exerciseState: {
+      index: 0,
+      playing: false,
+      timestamp: Timestamp.now(),
+    },
+    facilitator: 'some-user-id',
+    startTime: Timestamp.now(),
+    started: false,
+    ended: false,
+    userIds: ['all'],
+  },
+  {
+    id: 'some-other-session-id',
+    name: 'some-other-name',
+    url: 'some-other-url',
+    exerciseState: {
+      index: 0,
+      playing: false,
+      timestamp: Timestamp.now(),
+    },
+    facilitator: 'some-other-user-id',
+    startTime: Timestamp.now(),
+    started: false,
+    ended: false,
+    userIds: ['all'],
+  },
+];
+
+beforeEach(async () => {
+  await Promise.all(
+    sessions.map(session =>
+      firestore().collection('sessions').doc(session.id).set(session),
+    ),
+  );
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('session model', () => {
```

Comment:
@swemail I ended up transforming the old route specs into model specs, would love to get your input at some point, maybe there are better ways to test it (or not test it at all 🤷)

Reply:
Great! I think this is the only way test queries against the mock, great that you moved it over so we can have it as reference.

Reply:
I see that we've controllers now already, so I think i will move some of the logic into them, so the tests for models would be very straight-forward

Reply:
🥚 sactly, the models will be dumb, but it's not wrong to verify the queries so you kan keep that part of the test I think